### PR TITLE
US118647 - Replaced simple-overlay with d2l-dialog-fullscreen component in rubrics list

### PR DESCRIPTION
**Pull request by annabelsegalld2l**
The d2l-simple-overlay component is being replaced with the d2l-dialog-fullscreen component. The functionality of the page should not be affected. 

Here's a gif of the new look: 
add_rubric_new

I noticed that the "Edit rubric" title is part of the body, rather than being in the header. We can move that to the header, but I decided to keep it as it was for this PR(the overlay header was also empty).

### CODE REVIEW

File: components/d2l-activity-editor/d2l-activity-rubrics/d2l-activity-rubrics-list-container.js [hunk 1]
```diff
@@ -76,23 +76,17 @@ class ActivityRubricsListContainer extends ActivityEditorMixin(RtlMixin(Localize
 
 			${this._renderDefaultScoringRubric(entity)}
 
-			<d2l-simple-overlay
+			<d2l-dialog-fullscreen
 				id="create-new-association-dialog"
-				close-simple-overlay-alt-text="${this.localize('rubrics.btnClose')}"
```
Comment (svanherk) [inline components/d2l-activity-editor/d2l-activity-rubrics/d2l-activity-rubrics-list-container.js:81]: Looks like we can cleanup the `rubrics.btnClose` langterm

**Comment by svanherk** [comment]
> I noticed that the "Edit rubric" title is part of the body, rather than being in the header. We can move that to the header, but I decided to keep it as it was for this PR(the overlay header was also empty).

Huh that's interesting.  Looks like the "Edit Rubric" is attached to the rubric editor so maybe it was done because having the two titles would look silly, but it also looks like there's a slot that you could maybe override to hide the editor title.  The dialog titles is an `h2` versus this title is just a `div` so not sure the screen reader experience difference.  We also say the `title-text` is required on the dialog, so maybe @dbatiste can weigh in there...

**Comment by svanherk** [comment]
Is this used in FACE?  This looks good to me but getting someone from that project's eyes on this would be good as well

File: components/d2l-activity-editor/d2l-activity-rubrics/d2l-activity-rubrics-list-container.js [hunk 2]
```diff
@@ -76,23 +76,17 @@ class ActivityRubricsListContainer extends ActivityEditorMixin(RtlMixin(Localize
 
 			${this._renderDefaultScoringRubric(entity)}
 
-			<d2l-simple-overlay
+			<d2l-dialog-fullscreen
```
Comment (annabelsegalld2l) [inline components/d2l-activity-editor/d2l-activity-rubrics/d2l-activity-rubrics-list-container.js:79]: One thing we can do to fix the empty title is consider adding that no-header option. I know that was in development by Dave B but the PR was never merged.

**Comment by dbatiste** [comment]
> We also say the `title-text` is required on the dialog, so maybe @dbatiste can weigh in there...

Yeah, we should try to use the title - besides providing a more balanced layout, it labels the dialog element.  The a11y experience likely varies by browser + screen reader combination.

**Comment by annabelsegalld2l** [comment]
I added @mdulat from FACE since she is last the last commit to this file. Feel free to add more reviewers if needed Margaret.

**Review by mdulat — APPROVED** [review]
LGTM.
I think the reason the title is in the body is because it's part of the `d2l-rubric-editor` component, though it should be easy enough to leave out the header as an option. I believe it was also left in there because Design wanted it to be inline with the status, and we couldn't put the status into the dialog header. So maybe check with Design before moving it into the dialog header, it does seem like the better a11y solution.

**Review by mdulat — APPROVED** [review]

File: components/d2l-activity-editor/d2l-activity-rubrics/d2l-activity-rubrics-list-container.js [hunk 3]
```diff
@@ -76,23 +76,18 @@ class ActivityRubricsListContainer extends ActivityEditorMixin(RtlMixin(Localize
...
 				id="create-new-association-dialog"
-				close-simple-overlay-alt-text="${this.localize('rubrics.btnClose')}"
-				no-cancel-on-outside-click
-				@d2l-simple-overlay-close-button-clicked="${this._clearNewRubricHref}"
-				@d2l-simple-overlay-canceled="${this._clearNewRubricHref}"
-			>
+				title-text = ${this.localize('rubrics.hdrCreateRubric')}
```
Comment (svanherk) [inline components/d2l-activity-editor/d2l-activity-rubrics/d2l-activity-rubrics-list-container.js:81]: In the slack thread Patrick mentioned changing this name just for FACE... is `d2l-activity-rubrics-list-container` only used in FACE?  Or whatever uses that only used in FACE?

Comment (mdulat) [reply components/d2l-activity-editor/d2l-activity-rubrics/d2l-activity-rubrics-list-container.js:81]: Currently it's only used in FACE, yes.

Comment (annabelsegalld2l) [reply components/d2l-activity-editor/d2l-activity-rubrics/d2l-activity-rubrics-list-container.js:81]: Yep, this is d2l-activity-editor, which is only used by the new FACE experience.

**Review by svanherk — APPROVED** [review]

**Pull request merged**